tools/test/toolchains/test_toolchains.py

```diff
@@ -68,32 +68,73 @@ def test_arm_default_lib(self):
         self.assertIn("-Wl,--library_type=microlib", arm_c6_obj.flags["cxx"])
         self.assertIn("--library_type=microlib", arm_c6_obj.flags["asm"])
 
-    def test_arm_default_lib_std_exception(self):
+    def test_arm_c_lib_std_exception(self):
         """Test that an exception is raised if the std C library is not supported for a target on the ARM toolchain."""
         mock_target = mock.MagicMock()
         mock_target.core = "Cortex-M4"
         mock_target.supported_toolchains = ["ARM", "uARM", "ARMC5"]
         mock_target.default_toolchain = "ARM"
-        mock_target.default_lib = "std"
+        mock_target.c_lib = "std"
         mock_target.supported_c_libs = {"arm": ["small"]}
-        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.default_lib)):
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.c_lib)):
             ARM_STD(mock_target)
-        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.default_lib)):
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.c_lib)):
             ARMC6(mock_target)
 
 
-    def test_arm_default_lib_small_exception(self):
+    def test_arm_c_lib_small_exception(self):
         """Test that an exception is raised if the small C library is not supported for a target on the ARM toolchain."""
         mock_target = mock.MagicMock()
         mock_target.core = "Cortex-M4"
-        mock_target.default_lib = "small"
+        mock_target.c_lib = "small"
         mock_target.supported_c_libs = {"arm": ["std"]}
         mock_target.default_toolchain = "ARM"
         mock_target.supported_toolchains = ["ARM", "uARM", "ARMC5"]
-        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.default_lib)):
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.c_lib)):
             ARM_STD(mock_target)
         mock_target.default_toolchain = "ARMC6"
-        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.default_lib)):
+        with self.assertRaisesRegexp(NotSupportedException, UNSUPPORTED_C_LIB_EXCEPTION_STRING.format(mock_target.c_lib)):
+            ARMC6(mock_target)
+
+    def test_arm_selected_c_libs(self):
```

Comment:
The name is not very precise. I would use:
```suggestion
    def test_arm_small_lib_selection(self):
```